ospf6d/ospf6_asbr.c

```diff
@@ -935,18 +938,35 @@ static void ospf6_asbr_routemap_update(const char *mapname)
 			ospf6->rmap[type].map = route_map_lookup_by_name(
 				ospf6->rmap[type].name);
 
-			if (mapname && ospf6->rmap[type].map
+			if (mapname
 			    && (strcmp(ospf6->rmap[type].name, mapname) == 0)) {
-				if (IS_OSPF6_DEBUG_ASBR)
-					zlog_debug(
-						"%s: route-map %s update, reset redist %s",
-						__func__, mapname,
-						ZROUTE_NAME(type));
+				if (ospf6->rmap[type].map) {
+					if (IS_OSPF6_DEBUG_ASBR)
+						zlog_debug(
+							"%s: route-map %s update, reset redist %s",
+							__func__, mapname,
+							ZROUTE_NAME(type));
 
-				route_map_counter_increment(
-					ospf6->rmap[type].map);
+					route_map_counter_increment(
+						ospf6->rmap[type].map);
 
-				ospf6_asbr_distribute_list_update(type);
+					ospf6_asbr_distribute_list_update(type);
+				} else {
+					/*
+					 * if the mapname matches a route-map on
+					 * ospf6 but the map doesn't exist, it
+					 * is being deleted. flush and then
```

Comment:
LGTM. I think there are more fixes required which may be done in separate PR may be as this fixes delete route-map handling. 

Below is one of the example where it can be handled better. 

If I have prefix-list for permit any and all the static routes are redistributed and then same command is issued again then we send redistribute delete and add to zebra importing routes from zebra again while nothing has changed. 

2020/05/16 07:12:39 OSPF6: ospf6_asbr_distribute_list_update: trigger redistribute static reset thread
2020/05/16 07:12:42 OSPF6: sendmsg failed: ifindex: 8: Network is unreachable (101)
2020/05/16 07:12:42 OSPF6: [EC 100663307] Could not send entire message
2020/05/16 07:12:44 OSPF6: ospf6_asbr_routemap_update_timer: route-map R1 update, reset redist static
2020/05/16 07:12:44 OSPF6: Zebra Receive route add: static 2001::1/128 nexthop :: ifindex 4 tag 0
2020/05/16 07:12:44 OSPF6: Redistribute 2001::1/128 (static)
2020/05/16 07:12:44 OSPF6: Advertise as AS-External Id:0.0.0.16 prefix 2001::1/128 metric 0
2020/05/16 07:12:44 OSPF6: Originate AS-External-LSA for 2001::1/128
2020/05/16 07:12:44 OSPF6: Zebra Receive route add: static 3001::1/128 nexthop :: ifindex 4 tag 2000
2020/05/16 07:12:44 OSPF6: Redistribute 3001::1/128 (static)
2020/05/16 07:12:44 OSPF6: Advertise as AS-External Id:0.0.0.17 prefix 3001::1/128 metric 0
2020/05/16 07:12:44 OSPF6: Originate AS-External-LSA for 3001::1/128
2020/05/16 07:12:44 OSPF6: Zebra Receive route add: static 3001::4/128 nexthop :: ifindex 4 tag 2000